internal/validation/v2alpha1/hosts_test.go

```diff
@@ -0,0 +1,327 @@
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(0))
+	})
+
+	It("Should fail if host is not FQDN", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("host-without-domain"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(1))
+		Expect(problems[0].AttributePath).To(Equal(".spec.hosts[0]"))
+		Expect(problems[0].Message).To(Equal("Host is not fully qualified domain name"))
+	})
+
+	It("Should allow lenghty host name with numbers and dashes", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("host-with-numbers-123-and-dashes.long-domain-with-numbers-123-and-dashes.com"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(0))
+	})
+
+	It("Should fail if host name contains wrong characters", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("host-with-wrong-character-like-*.example.com"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(1))
+		Expect(problems[0].AttributePath).To(Equal(".spec.hosts[0]"))
+		Expect(problems[0].Message).To(Equal("Host is not fully qualified domain name"))
+	})
+
+	It("Should fail if any segment in host name starts or ends with dash", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("-host.example.com"),
+					getHostPtr("host-.example.com"),
+					getHostPtr("host.example-.com"),
+					getHostPtr("host.example.-com"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(4))
+		for i := 0; i < 4; i++ {
+			Expect(problems[i].AttributePath).To(Equal(fmt.Sprintf(".spec.hosts[%d]", i)))
+			Expect(problems[i].Message).To(Equal("Host is not fully qualified domain name"))
+		}
+	})
+
+	It("Should fail if any host is not FQDN", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("host.example.com"),
+					getHostPtr("host-without-domain"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(1))
+		Expect(problems[0].AttributePath).To(Equal(".spec.hosts[1]"))
+		Expect(problems[0].Message).To(Equal("Host is not fully qualified domain name"))
+	})
+
```

Comment:
please add/cover all tests for the length limitations on FQDN validation:
```

const (
	fqdnMaxLength        = 253
	fqdnMinSegments      = 2
	domainLabelMaxLength = 63
)
```

Reply:
I've just added much more tests covering all these cases